### DISCUSSION TITLE
Accessing cache metadata?

### DISCUSSION THREAD

**Discussion by deadowl** [smarty-php/smarty#901] [Ideas]
I've got smarty code that loads dynamically into Wordpress based on URL rewriting (works as a single page in WP). I want to generate an XML sitemap for the dynamic content, ideally with  lastmod element included. Cache key is a serialized 2 element associative array. I'd clearly have to change some other code to clear the cache using Smarty's methods. Meanwhile, there doesn't seem to be anything revealing filemtime from the internal cache handler. Any thoughts or ideas?

**Comment by wisskid** [comment]
First of all, it seems wrong to use the datetime of the cache as the source for a last modified field. Caches may be deleted at any point (e.g. during maintenance or when you upgrade Smarty) and one wouldn't expect the dates in the sitemap to change because of that.

That said, would it be feasible to implement a custom cache handler that wraps Smarty's default implementation and adds an extra method that returns the datetime when te cache was stored?

**Reply by deadowl** [reply]
Thanks for the input. I was thinking of doing that, if maybe the words in the documentation on cache clearing "would be like removing the files" might imply there's something a little more going on behind the scenes, such as only deleting cached files when there are actual changes? I'm unfamiliar with opcache_invalidate and apc_delete_file, but I'm also seeing unlink and rmdir in there. My source data is from a bunch of database tables with their own activity dates which would take a bit of time to reconcile and wouldn't necessarily be maintainable if an incomplete picture of the data is being used in the templates, and then of course have to dynamically account for the Wordpress backdrop data changing as well. I'm thinking I'll probably just have to settle for not including lastmod in the generated sitemap. Thank you for taking the time to respond to my question. I very much appreciate your work on this project.
